libs/data/bbhnet/data/dataloader.py

```diff
@@ -1,476 +1,40 @@
-                `waveform_frac` must be set to 0.
-            waveform_frac:
-                The fraction of each batch that should consist
-                of injected waveforms, and be marked with a
-                `1.` in the target tensor produced during iteration.
-            min_snr:
-                Minimum SNR value for sampled waveforms. Must be
-                specified if `waveform_dataset` is not `None`.
-            max_snr:
-                Maximum SNR value for sampled waveforms. Must be
-                specified if `waveform_dataset` is not `None`.
-            glitch_sampler:
-                Object for sampling glitches for insertion (not injection)
-                into background samples at data loading time. Must
-                include a `.sample` method. If passed a string, should be
-                a path to an HDF5 file containing glitches from each
-                interferometer contained in `"H1_glitches"` and
-                `"L1_glitches"` dataset keys. Data should be
-                sampled at rate `sample_rate`. Glitch triggers
-                should sit in the middle of the time axis of
-                the array containing the glitch timeseries.
-                Glitches will be randomly sampled and inserted in
-                place of the corresponding interferometer channel
-                background at data-loading time. If left as `None`,
-                no glitches will be inserted at data loading time
-                and `glitch_frac` should be 0.
-            glitch_frac:
-                The fraction of each batch that should consist
-                of inserted glitches, marked as `0.` in the
-                target tensor produced during iteration
-            trigger_distance:
-                The maximum number of seconds the t0 of signals and glitches
-                can lie away from the edge of sampled kernels
-        """
-
-        # default behavior is set trigger_distance to half
-        # kernel length. With this setting t0 can lie anywhere
-        # in the kernel
-        self.trigger_distance_size = trigger_distance * sample_rate
-
-        # sanity check our fractions
-        assert 0 <= waveform_frac <= 1
-        assert 0 <= glitch_frac <= 1
-
-        # initialize our device to be cpu so that we
-        # have to be explicit about forcing the background
-        # tensors to the device at lower precision
-        self.device = "cpu"
-        self.sample_rate = sample_rate
-        self.kernel_size = int(kernel_length * sample_rate)
-        self.batch_size = batch_size
-        self.batches_per_epoch = batches_per_epoch
-
-        # load in the background data
-        self.hanford_background = _load_background(hanford_background, frac)
-        self.livingston_background = _load_background(
-            livingston_background, frac
+        super().__init__(
+            X,
+            kernel_size,
+            batch_size=batch_size,
+            stride=1,

```

Comment:
Ah ok, I think my mistake was actually probably in including this in the arguments to the dataloader, since my assumption was that we'll always want a stride of 1 for this dataloader, but open to thoughts here: correct this to be `stride=stride`, or remove the `stride` argument altogether from the `BBHNetInMemoryDataloader`?

Reply:
Remind me: a `stride` of 1 means the dataloader will move by 1 `sample` when creating `kernels` of background data? If so, I agree. We'll probably only ever want strides of 1.

Reply:
Yeah exactly, or in this case that it's capable of sampling kernels at a resolution of 1 sample. But I agree, I'll take the arg out of the dataloader then.